src/main/java/net/pms/configuration/RendererConfiguration.java

```diff
@@ -1251,10 +1254,17 @@ public boolean isTranscodeAudioTo441() {
 	}
 
 	/**
-	 * @return whether to transcode H.264 video if it exceeds level 4.1
+	 * @return The H.264 level limit, if any, or {@code null}.
 	 */
-	public boolean isH264Level41Limited() {
-		return getBoolean(H264_L41_LIMITED, true);
+	public H264Level getH264LevelLimit() {
+		return H264Level.typeOf(getString(H264_LEVEL_LIMIT, null));
```

Comment:
Maybe it should default to level 4.1?

Reply:
@SubJunk agree that will be better to set the default value to level 4.1 and in this case the implemented logic in the DLNAResource line  976 must be changed to
 `} else if (media.isH264() && renderer.getH264LevelLimit() != H264Level.L4_1) {` 
and for all renderers where previously was set `H264Level41Limited = false` and this setting was deleted in this PR should be added `H264LevelLimit = 5.2`.

Reply:
I'm not sure about this. Even though the old renderers were often limited to 4.1, is that really the case these days? I don't like to define defaults that isn't "logical", if no limit is defined for a given renderer it's most logical to me that there actually isn't a limit for that renderer.

Reply:
@Nadahar I am not sure either but for me this is the safest way how to implement this PR. Users are using the UMS with the restriction to level 4.1 for the long time without complaining except users with renderers where was set `H264Level41Limited = false`. What we offer in this PR is only a different way how to decide which video will be streamed or transcoded.

Reply:
Maybe this isn't the case anymore but AFAIK most renderers support 4.1 nicely. As @ExSport said in https://github.com/UniversalMediaServer/UniversalMediaServer/issues/1417#issuecomment-346393516 it can cause real problems the other way too (and it did) so either way we will inconvenience some users.
I think 4.1 is a safer default than blank though because it's better to have transcoded video than no video.

Reply:
@SubJunk by the way we use the function `isVideoWithinH264LevelLimits` mentioned by @ExSport only in FFmpegVideo, MencoderVideo and TsMuxeRVideo for recognition if the video is correct to be muxed not during media parsing to distinguish if the media will be streamed or transcoded. I tried to use it also for parsing but it slows down the process because it calls the function `getAnnexBFrameHeader` where the FFmpeg is used to obtain the `h264_annexB` header and it takes long time.

Reply:
@SubJunk I think it's better if playback fails than if transcoding is forced when not needed. If it fails, we have a chance to find out (because they post that it doesn't work) and correct the renderer configuration. If it transcodes when not needed, it's unlikely that the user will report it, and they will just be annoyed by the delay, the lack of seeking, the lack of audio and subtitles choices on the renderer and the quality loss. 

It's fine that most renderers support 4.1 - the problem is that many can take more.

Reply:
@Nadahar I didn't look at it from this angle and must say that you are right.

Reply:
I don't agree with that, in fact our top competitor Plex aggressively transcodes on many renderers - they don't even have many specific renderer profiles. I think if a user has broken playback they will usually just uninstall us and go find something that works. Sometimes we get users reporting these things but that's not something to rely on.

Anyway I'm outvoted so go ahead :)

Reply:
I essentially only logged on to reduce my mail spam by unwatching, but couldn'thelp myself. This must be one of the stupidest things I've ever heard when working with software design. As @SubJunk  says normally if playback fails the user will just say "oh this doen't work". He won't write a bug report, since he will consider the program broken (failing to perform it's main task). He is more likely to write a report if seeking fails since that is a feature.

Reply:
@SubJunk It's nothing that I feel that strongly about, I was just explaining my stance and why I did like I did. I've seen many examples on people being annoyed by needless transcoding. Some will probably turn away in either case, which is why this should be set correctly - not rely on the default.

One thing that also weighs in for me and that probably only is a "me thing" is that I feel it is "logically wrong" to set a default that is just a "random guess". I value "quality of information", so I like to make sure that whatever information I have is reasonably accurate, and prefer knowing that I don't know over using a guess that seems like real information to me. I realize that this is very likely to be something that is just important to me, and that the users won't care about this at all.

When it comes to Plex I don't really look at what they are doing. I don't consider Plex a "competitor" since it's payware. I tried the free version for a little bit a long time ago, and it annoyed the hell out of me with all it's lack of configuration/decisions being made for you. I know that for all the Apple users out there, this is appealing - but I think it's important to note what most companies seems to miss today as they rush to copy Apple - that this isn't the whole market. It's the most ignorant part of the market that appreciates having no choices, as they don't know how to make the decisions and rely 100% on the default behaviour. However, given how "difficult" UMS/PMS is and has always been, I doubt that's our current user base. If you just cater to the ignorant you will alienate the informed. That said, this is a general view of mine, that isn't particularly relevant to this decision, but it explains why I never look to what Plex does. For all I care, they cater to a different user base.

@SharkHunter Can I do anything to help you unwatch? I honestly can't wait to get rid of your mostly stupid and useless comments. I have a very hard time believing that this is one of the stupidest things you've ever heard, as you most often say more stupid things yourself.

Your argument doesn't really hold water - the default will "alienate" some users whatever choice is made. You're right that most users won't report, but the question is what will affect most users and how bad. If most renderers can play a higher level so that a lot of users will be impacted by needless transcoding and all the negative aspects with that, I'd say the average user experience is worse limiting to 4.1. If most renderers can't play a higher level, the situation is the opposite. In addition, my argument was that for those few that do report issues, it's much harder to understand what's wrong if the default is too low - so the chance that we will ever know about it and be able to correct it is much smaller.

Reply:
My assumption is that most users don't even know what transcoding is. They are just installing a program that serves video to their device. If they don't see video on the device then our program is broken. If they see it, it works.
Our advanced users are capable of seeing that there is transcoding happening and maybe they don't like hearing the fan in the computer make more noise than usual or slower seeking, but as advanced users they can be the ones who submit the better renderer configs to us to reduce the transcoding.

So from my perspective we will make the most people happy by succeeding at playing video more often.

About the comment "given how difficult UMS/PMS is", you're right but it has always been my goal to make it easier for new users and we have made good progress on that. I hope we make more so that we can still offer the advanced features to opt-in to while "just working" for our Plex-lovers :)